{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# Generage figres from Micro-C data.\n",
    "- This script uses the coolbox api along with data generates with micro-c, cooler, mustache, and juicer to plot regions of the Drosophila genome with differences in chromatin.  \n",
    "### To create the conda environment for this script:\n",
    "\n",
    "1. mamba create env -n coolbox -c bioconda coolbox \n",
    "2. mamba activate coolbox\n",
    "3. mamba install -c anaconda ipykernel\n",
    "4. python -m ipykernel install --user --name=coolbox\n",
    "\n",
    "Then select the coolbox kernel under python environments within the notebook.\n",
    "- https://medium.com/@nrk25693/how-to-add-your-conda-environment-to-your-jupyter-notebook-in-just-4-steps-abeab8b8d084"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "9/27/24 Note to self: \n",
    "\n",
    "- Use the mustache v4 datasets for plotting the diffloops and gtf's. The files have already been generated and only loops called by mustache across all pairwise comparisons are in those combined datasets. This will allow me to plot the merged maps for JW18 DOX and wMel with the filtered gtf's and the diffloops. \n",
    "\n",
    "- I want to change how the plots look so I can:\n",
    "1. Search by gene \n",
    "2. Plot the two contact maps one on top the other inverted, show the p-values too. \n",
    "3. Draw circles at tips instead of triangles across the call.\n",
    "\n",
    "10/1/24 Note to self:\n",
    "Try plotting diffloop points with HiCPeaksCoverage(loops1, color=\"#2255ff\", line_width=5), this needs a bedpe file that you need to generate. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Current working directory: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import coolbox\n",
    "from coolbox.api import *\n",
    "import pandas as pd\n",
    "from math import floor\n",
    "\n",
    "# change working directory\n",
    "os.chdir(\"/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map\")\n",
    "print(f\"Current working directory: {os.path.abspath(os.curdir)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "pairwise = {\n",
    "  'JW18-DOX_JW18-wMel' : \t['JW18-DOX',\t'JW18-wMel'],\n",
    " }\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "Files:\n",
    "1. /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/JW18-DOX.matrix_1kb.mcool \n",
    "2. /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/JW18-wMel.matrix_1kb.mcool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot(TEST_RANGE,pairwise,comparison,resolution,directory):\n",
    "    #Contact map 1\n",
    "    file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "    #Contact map2\n",
    "    file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "    #genes with open regions that are enriched in file1\n",
    "    gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "    #genes with open regions that are enriched in file1\n",
    "    gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "    #mustache identified de loops1\n",
    "    loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "    #mustache identified de loops2\n",
    "    loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "    \n",
    "    c1,c2=pairwise[comparison]\n",
    "\n",
    "    with MinValue(-9), MaxValue(-1):\n",
    "        cool1 = Cool(file1)\n",
    "        cool2 = Cool(file2)\n",
    "\n",
    "    frame = ChromName()+ XAxis() + \\\n",
    "        Selfish(cool2, cool1, norm=\"log\") +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "        MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "        HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=\"bwr_r\") + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "        MinValue(-6) + MaxValue(6)  + \\\n",
    "        TADCoverage(loops1, border_only=False, alpha=0.25, color='red',border_color='red') +\\\n",
    "        TADCoverage(loops2, border_only=False, alpha=0.25, color='blue',border_color='blue') +\\\n",
    "        GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "        GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "        XAxis()\n",
    "    frame *= Feature(depth_ratio=0.35)\n",
    "    frame.plot(TEST_RANGE)\n",
    "\n",
    "    bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "    bsr.goto(TEST_RANGE)\n",
    "    bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Browser for given ranges:\n",
    "# comparison = 'JW18-DOX-1_JW18-wMel-1' \n",
    "# resolution = '10kb'\n",
    "# TEST_RANGE = '3L:10000-1000000'\n",
    "\n",
    "\n",
    "# file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "#     #Contact map2\n",
    "# file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "# #mustache identified de loops1\n",
    "# loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "# #mustache identified de loops2\n",
    "# loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "# c1,c2=pairwise[comparison]\n",
    "\n",
    "# with MinValue(-9), MaxValue(-1):\n",
    "#     cool1 = Cool(file1)\n",
    "#     cool2 = Cool(file2)\n",
    "\n",
    "# frame = ChromName()+ XAxis() + \\\n",
    "#     Selfish(cool2, cool1, norm=\"log\") +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "#     MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "#     HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=\"bwr_r\") + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "#     MinValue(-6) + MaxValue(6)  + \\\n",
    "#     TADCoverage(loops1, border_only=False, alpha=0.25, color='red',border_color='red') +\\\n",
    "#     TADCoverage(loops2, border_only=False, alpha=0.25, color='blue',border_color='blue') +\\\n",
    "#     GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "#     GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "#     XAxis()\n",
    "# frame *= Feature(depth_ratio=0.35)\n",
    "# frame.plot(TEST_RANGE)\n",
    "\n",
    "# bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "# bsr.goto(TEST_RANGE)\n",
    "# # bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib.colors import LinearSegmentedColormap\n",
    "\n",
    "# # give a list of color (color gradient)\n",
    "# uberplant = '#171123'\n",
    "# eggplants = '#372248'\n",
    "# periptant = '#414770'\n",
    "# blueplant = '#5B85AA'\n",
    "# kindablue = '#A9BCD0'\n",
    "fullwhite = '#FFFFFF'\n",
    "# sadorange = '#F9C784'\n",
    "# tangerine = '#FCAF58'\n",
    "# prforange = '#FF8C42'\n",
    "# xxxorange = '#F46036'\n",
    "# rustyredy = '#C2320A'\n",
    "\n",
    "# diff_gradient = [uberplant, eggplants, periptant, blueplant, kindablue, fullwhite, sadorange, tangerine, prforange, xxxorange, rustyredy]\n",
    "# selfish_gradient = [uberplant, eggplants, periptant, blueplant, kindablue, fullwhite]\n",
    "\n",
    "purpl = '#241e4e'\n",
    "blue1 = '#587792'\n",
    "blue2 = '#8db1ab'\n",
    "green = '#C7E576'\n",
    "yell1 = '#f2ed6f'\n",
    "yell2 = '#f4e04d'\n",
    "yell3 = '#F0D719'\n",
    "orang = '#ff4000'\n",
    "\n",
    "diff_gradient = [purpl,blue1, blue2, green, fullwhite, yell1, yell2, yell3, orang]\n",
    "selfish_gradient =[purpl, fullwhite]\n",
    "\n",
    "diff_cmap = LinearSegmentedColormap.from_list('interaction', diff_gradient)\n",
    "diff_cmap.set_bad(orang)\n",
    "diff_cmap.set_under(purpl)\n",
    "\n",
    "\n",
    "selfish_cmap = LinearSegmentedColormap.from_list('interaction', selfish_gradient)\n",
    "selfish_cmap.set_bad(orang)\n",
    "selfish_cmap.set_under(purpl)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Browser for given ranges:\n",
    "# directory = '/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation'\n",
    "# comparison = 'JW18-DOX_JW18-wMel' \n",
    "# resolution = '1kb'\n",
    "# TEST_RANGE = '3L:2980000-3970000'\n",
    "# # TEST_RANGE = '3R:31491078-31492678' \n",
    "# JW18_DOX  = '#BAD86E'\n",
    "# JW18_wMel = purpl\n",
    "# #contact map 1\n",
    "# file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "# #Contact map2\n",
    "# file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "# #mustache identified de loops1\n",
    "# loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "# #mustache identified de loops2\n",
    "# loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "\n",
    "# ###Tracks\n",
    "# #RNA-coverage \n",
    "# rnaseq_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18DOX221117.avg.coverage.bw'\n",
    "# rnaseq_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117.avg.coverage.bw'\n",
    "# #Micro-C_coverage \n",
    "# microc_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-DOX-1.coverage.bw'\n",
    "# microc_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel-1.coverage.bw'\n",
    "# #Mappability \n",
    "# mappability =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/dmel_6_genmap_e0_k150_final.bw'\n",
    "\n",
    "# c1,c2=pairwise[comparison]\n",
    "\n",
    "\n",
    "# with MinValue(-9), MaxValue(-1):\n",
    "#     cool1 = Cool(file1)\n",
    "#     cool2 = Cool(file2)\n",
    "\n",
    "# frame = ChromName()+ XAxis() + \\\n",
    "#     Selfish(cool2, cool1, norm=\"log\", cmap=selfish_cmap) +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "#     MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "#     HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=diff_cmap) + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "#     MinValue(-6) + MaxValue(6)  + \\\n",
    "#     TADCoverage(loops1, border_only=False, alpha=0.45, color=JW18_DOX,border_color=JW18_DOX) +\\\n",
    "#     TADCoverage(loops2, border_only=False, alpha=0.25, color=JW18_wMel,border_color=JW18_wMel) +\\\n",
    "#     GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "#     GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "#     BigWig(rnaseq_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-DOX-1\")+\\\n",
    "#     BigWig(rnaseq_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-wMel-1\")+\\\n",
    "#     BigWig(microc_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-DOX-1\")+\\\n",
    "#     BigWig(microc_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-wMel-1\")+\\\n",
    "#     BigWig(mappability) + Color('grey') + MinValue(0) + MaxValue(1)+ Title(\"Mappability\")+TrackHeight(1) +\\\n",
    "#     XAxis()\n",
    "# frame *= Feature(depth_ratio=0.35)\n",
    "# frame.plot(TEST_RANGE)\n",
    "\n",
    "# bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "# bsr.goto(TEST_RANGE)\n",
    "# bsr.show()\n",
    "# bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0559d4c01d2f40b98970c3c14c6d64f1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(VBox(children=(HBox(children=(Dropdown(options=('2L', '2R', '3L', '3R', '4', 'X', 'Y'), value='…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File saved to: /scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/JW18-DOX_JW18-wMel_2L:7600000-10000000.svg\n",
      "Gene 'FBgn0005677' not found in the GTF file.\n"
     ]
    }
   ],
   "source": [
    "# Browser for given ranges:\n",
    "directory = '/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation'\n",
    "comparison = 'JW18-DOX_JW18-wMel' \n",
    "resolution = '1kb'\n",
    "TEST_RANGE = '2L:7600000-10000000'\n",
    "# TEST_RANGE = '3R:31491078-31492678' \n",
    "JW18_DOX  = '#BAD86E'\n",
    "JW18_wMel = purpl\n",
    "#contact map 1\n",
    "file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "#Contact map2\n",
    "file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "#genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#all genes\n",
    "gtf_all='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-r6.46.sorted.uniq.gtf'\n",
    "# # #genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "#Genes enriched in both \n",
    "gtf3=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop.symbol.gtf'\n",
    "\n",
    "\n",
    "#mustache identified de loops1\n",
    "loops1='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/combined/diffloop1.bed'\n",
    "loops1_1kb = '/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/1kb/diffloop1.plot.bed'\n",
    "loops1_16kb = '/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/16kb/diffloop1.bed'\n",
    "loops1_bedpe ='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/1kb/diffloop1.bedpe'\n",
    "loops2_bedpe ='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/1kb/diffloop2.bedpe'\n",
    "#mustache identified de loops2\n",
    "loops2='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/combined/diffloop2.bed'\n",
    "loops2_1kb = '/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/1kb/diffloop2.plot.bed'\n",
    "loops2_16kb = '/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v4/16kb/diffloop2.bed'\n",
    "\n",
    "# print(f'cat {gtf1} {gtf2} |uniq > {gtf3}')\n",
    "\n",
    "###Tracks\n",
    "#RNA-coverage \n",
    "rnaseq_cov =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117_vs_DOX.diff.coverage.bw'\n",
    "# rnaseq_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18DOX221117.avg.coverage.bw'\n",
    "# rnaseq_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117.avg.coverage.bw'\n",
    "#Micro-C_coverage \n",
    "microc_cov =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel_vs_DOX.diff.coverage.bw'\n",
    "# microc_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-DOX-1.coverage.bw'\n",
    "# microc_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel-1.coverage.bw'\n",
    "#Mappability \n",
    "mappability =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/dmel_6_genmap_e0_k150_final.bw'\n",
    "\n",
    "c1,c2=pairwise[comparison]\n",
    "\n",
    "frame = ChromName()+ XAxis() + \\\n",
    "    Cool(file1, cmap='JuiceBoxLike') + Title(f\"{pairwise[comparison][0]} Contact Map\") + TrackHeight(12) +\\\n",
    "    HiCPeaksCoverage(loops1_bedpe, color='blue',line_width=5) +\\\n",
    "    Cool(file2, cmap='JuiceBoxLike',orientation='inverted') + Title(f\"{pairwise[comparison][1]} Contact Map\") + TrackHeight(12) +\\\n",
    "    HiCPeaksCoverage(loops2_bedpe, color='blue',line_width=5) +\\\n",
    "    Spacer(height=0.5) + \\\n",
    "    GTF(gtf_all)+Title(f\"Genetic elements\")+ TrackHeight(6)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(rnaseq_cov) + Color('red') + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\nJW18-wMel enrichment\\nover JW18-DOX\")+TrackHeight(0.5)+\\\n",
    "    BigWig(rnaseq_cov) + Color('black') + MinValue(-500) + MaxValue(0)+ TrackHeight(0.5)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(microc_cov) + Color('red') + MinValue(0) + MaxValue(1000)+ Title(\"Micro-C coverage\\nJW18-wMel enrichment\\nover JW18-DOX\")+TrackHeight(0.5)+\\\n",
    "    BigWig(microc_cov) + Color('black') + MinValue(-1000) + MaxValue(0)+ TrackHeight(0.5)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(mappability) + Color('grey') + MinValue(0) + MaxValue(1)+ Title(\"Mappability\")+TrackHeight(0.5) +\\\n",
    "    XAxis()\n",
    "frame *= Feature(depth_ratio=0.35)\n",
    "frame.plot(TEST_RANGE)\n",
    "\n",
    "bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "bsr.goto(TEST_RANGE)\n",
    "bsr.show()\n",
    "bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.svg')\n",
    "print(f\"File saved to: {directory}/{comparison}_{TEST_RANGE}.svg\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ChatGPT Modifications on 9/30/24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene 'FBgn0000097' coordinates: 2L:2056484-2278754\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0559d4c01d2f40b98970c3c14c6d64f1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(VBox(children=(HBox(children=(Dropdown(options=('2L', '2R', '3L', '3R', '4', 'X', 'Y'), value='…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File saved to: /scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/JW18-DOX_JW18-wMel_2L:2056484-2278754.svg\n"
     ]
    }
   ],
   "source": [
    "# Search browser for specific Flybase ID\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "# Function to search for gene coordinates in the GTF file using gene_id\n",
    "def get_gene_range_by_id(gtf_file, gene_id, buffer=100000):\n",
    "    \"\"\"\n",
    "    Searches for a given gene in a GTF file by gene_id and returns its chromosomal range.\n",
    "\n",
    "    Parameters:\n",
    "    gtf_file : str\n",
    "        Path to the GTF file.\n",
    "    gene_id : str\n",
    "        Gene ID to search for (e.g., \"FBgn0031208\").\n",
    "    buffer : int\n",
    "        Additional range to include around the gene coordinates (default=10000).\n",
    "\n",
    "    Returns:\n",
    "    str\n",
    "        Chromosomal coordinates in 'chr:start-end' format or None if not found.\n",
    "    \"\"\"\n",
    "    # Load the GTF file into a DataFrame\n",
    "    df = pd.read_csv(gtf_file, sep='\\t', comment='#', header=None, names=[\n",
    "        'seqname', 'source', 'feature', 'start', 'end', 'score', 'strand', 'frame', 'attribute'])\n",
    "    \n",
    "    # Extract the gene_id from the attribute column\n",
    "    df['gene_id'] = df['attribute'].apply(lambda x: x.split('gene_id \"')[1].split('\"')[0] if 'gene_id \"' in x else None)\n",
    "\n",
    "    # Search for the gene using the gene_id\n",
    "    gene_info = df[df['gene_id'] == gene_id]\n",
    "    if not gene_info.empty:\n",
    "        # Get chromosomal range with buffer applied\n",
    "        start = max(gene_info['start'].min() - buffer, 0)  # Avoid negative coordinates\n",
    "        end = gene_info['end'].max() + buffer\n",
    "        chrom = gene_info['seqname'].unique()[0]  # Assume all records for a gene have the same seqname\n",
    "        return f'{chrom}:{start}-{end}'\n",
    "    else:\n",
    "        print(f\"Gene ID '{gene_id}' not found in the GTF file.\")\n",
    "        return None\n",
    "\n",
    "# Example usage for gene search by gene_id\n",
    "gtf_file = gtf_all\n",
    "gene_id = \"FBgn0000097\"  # Specify the gene_id you want to search\n",
    "gene_range = get_gene_range_by_id(gtf_file, gene_id)\n",
    "\n",
    "if gene_range:\n",
    "    print(f\"Gene '{gene_id}' coordinates: {gene_range}\")\n",
    "else:\n",
    "    print(\"No coordinates found for the specified gene ID.\")\n",
    "\n",
    "if gene_range:\n",
    "    bsr.goto(gene_range)\n",
    "    bsr.show()\n",
    "    bsr.save(f'{directory}/{comparison}_{gene_range}.svg')\n",
    "    print(f\"File saved to: {directory}/{comparison}_{gene_range}.svg\")\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "coolbox",
   "language": "python",
   "name": "coolbox"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
