{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# Generage figres from Micro-C data.\n",
    "- This script uses the coolbox api along with data generates with micro-c, cooler, mustache, and juicer to plot regions of the Drosophila genome with differences in chromatin.  \n",
    "### To create the conda environment for this script:\n",
    "\n",
    "1. mamba create env -n coolbox -c bioconda coolbox \n",
    "2. mamba activate coolbox\n",
    "3. mamba install -c anaconda ipykernel\n",
    "4. python -m ipykernel install --user --name=coolbox\n",
    "\n",
    "Then select the coolbox kernel under python environments within the notebook.\n",
    "- https://medium.com/@nrk25693/how-to-add-your-conda-environment-to-your-jupyter-notebook-in-just-4-steps-abeab8b8d084"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Current working directory: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import coolbox\n",
    "from coolbox.api import *\n",
    "import pandas as pd\n",
    "from math import floor\n",
    "\n",
    "# change working directory\n",
    "os.chdir(\"/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map\")\n",
    "print(f\"Current working directory: {os.path.abspath(os.curdir)}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pairwise={\n",
    "    'JW18-DOX-1_JW18-DOX-2':\t['JW18-DOX-1',\t'JW18-DOX-2'],\n",
    "    'JW18-DOX-1_JW18-wMel-1':\t['JW18-DOX-1',\t'JW18-wMel-1'],\n",
    "    'JW18-DOX-1_JW18-wMel-2':\t['JW18-DOX-1',\t'JW18-wMel-2'],\n",
    "    'JW18-DOX-2_JW18-wMel-1':\t['JW18-DOX-2',\t'JW18-wMel-1'],\n",
    "    'JW18-DOX-2_JW18-wMel-2':\t['JW18-DOX-2',\t'JW18-wMel-2'],\n",
    "    'JW18-DOX-1_JW18-wRi-1':\t['JW18-DOX-1',\t'JW18-wRi-1'],\n",
    "    'JW18-DOX-1_JW18-wRi-2':    ['JW18-DOX-1',\t'JW18-wRi-2'],\n",
    "    'JW18-DOX-2_JW18-wRi-1':\t['JW18-DOX-2',\t'JW18-wRi-1'],\n",
    "    'JW18-DOX-2_JW18-wRi-2':\t['JW18-DOX-2',\t'JW18-wRi-2'],\n",
    "    'JW18-DOX-1_JW18-wWil-1':\t['JW18-DOX-1',\t'JW18-wWil-1'],\n",
    "    'JW18-DOX-1_JW18-wWil-2':\t['JW18-DOX-1',\t'JW18-wWil-2'],\n",
    "    'JW18-DOX-2_JW18-wWil-1':\t['JW18-DOX-2',\t'JW18-wWil-1'],\n",
    "    'JW18-DOX-2_JW18-wWil-2':\t['JW18-DOX-2',\t'JW18-wWil-2'],\n",
    "    'JW18-wMel-1_JW18-wRi-1':\t['JW18-wMel-1',\t'JW18-wRi-1'],\n",
    "    'JW18-wMel-1_JW18-wRi-2':\t['JW18-wMel-1',\t'JW18-wRi-2'],\n",
    "    'JW18-wMel-2_JW18-wRi-1':\t['JW18-wMel-2',\t'JW18-wRi-1'],\n",
    "    'JW18-wMel-2_JW18-wRi-2':\t['JW18-wMel-2',\t'JW18-wRi-2'],\n",
    "    'JW18-wMel-1_JW18-wWil-1':\t['JW18-wMel-1',\t'JW18-wWil-1'],\n",
    "    'JW18-wMel-1_JW18-wWil-2':\t['JW18-wMel-1',\t'JW18-wWil-2'],\n",
    "    'JW18-wMel-2_JW18-wWil-1':\t['JW18-wMel-2',\t'JW18-wWil-1'],\n",
    "    'JW18-wMel-2_JW18-wWil-2':\t['JW18-wMel-2',\t'JW18-wWil-2']\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot(TEST_RANGE,pairwise,comparison,resolution,directory):\n",
    "    #Contact map 1\n",
    "    file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "    #Contact map2\n",
    "    file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "    #genes with open regions that are enriched in file1\n",
    "    gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "    #genes with open regions that are enriched in file1\n",
    "    gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "    #mustache identified de loops1\n",
    "    loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "    #mustache identified de loops2\n",
    "    loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "    \n",
    "    c1,c2=pairwise[comparison]\n",
    "\n",
    "    with MinValue(-9), MaxValue(-1):\n",
    "        cool1 = Cool(file1)\n",
    "        cool2 = Cool(file2)\n",
    "\n",
    "    frame = ChromName()+ XAxis() + \\\n",
    "        Selfish(cool2, cool1, norm=\"log\") +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "        MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "        HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=\"bwr_r\") + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "        MinValue(-6) + MaxValue(6)  + \\\n",
    "        TADCoverage(loops1, border_only=False, alpha=0.25, color='red',border_color='red') +\\\n",
    "        TADCoverage(loops2, border_only=False, alpha=0.25, color='blue',border_color='blue') +\\\n",
    "        GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "        GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "        XAxis()\n",
    "    frame *= Feature(depth_ratio=0.35)\n",
    "    frame.plot(TEST_RANGE)\n",
    "\n",
    "    bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "    bsr.goto(TEST_RANGE)\n",
    "    bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[WARNING:plot.py:486 -            plot_tads()] No regions found for Coverage TAD.1.\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[WARNING:plot.py:486 -            plot_tads()] No regions found for Coverage TAD.1.\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/10kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n"
     ]
    }
   ],
   "source": [
    "# Browser for given ranges:\n",
    "comparison = 'JW18-DOX-1_JW18-wMel-1' \n",
    "resolution = '10kb'\n",
    "TEST_RANGE = '3L:10000-1000000'\n",
    "\n",
    "\n",
    "file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "    #Contact map2\n",
    "file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "#genes with open regions that are enriched in file1\n",
    "gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#genes with open regions that are enriched in file1\n",
    "gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "#mustache identified de loops1\n",
    "loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "#mustache identified de loops2\n",
    "loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "c1,c2=pairwise[comparison]\n",
    "\n",
    "with MinValue(-9), MaxValue(-1):\n",
    "    cool1 = Cool(file1)\n",
    "    cool2 = Cool(file2)\n",
    "\n",
    "frame = ChromName()+ XAxis() + \\\n",
    "    Selfish(cool2, cool1, norm=\"log\") +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "    MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "    HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=\"bwr_r\") + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "    MinValue(-6) + MaxValue(6)  + \\\n",
    "    TADCoverage(loops1, border_only=False, alpha=0.25, color='red',border_color='red') +\\\n",
    "    TADCoverage(loops2, border_only=False, alpha=0.25, color='blue',border_color='blue') +\\\n",
    "    GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "    GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "    XAxis()\n",
    "frame *= Feature(depth_ratio=0.35)\n",
    "frame.plot(TEST_RANGE)\n",
    "\n",
    "bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "bsr.goto(TEST_RANGE)\n",
    "# bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib.colors import LinearSegmentedColormap\n",
    "\n",
    "# # give a list of color (color gradient)\n",
    "# uberplant = '#171123'\n",
    "# eggplants = '#372248'\n",
    "# periptant = '#414770'\n",
    "# blueplant = '#5B85AA'\n",
    "# kindablue = '#A9BCD0'\n",
    "fullwhite = '#FFFFFF'\n",
    "# sadorange = '#F9C784'\n",
    "# tangerine = '#FCAF58'\n",
    "# prforange = '#FF8C42'\n",
    "# xxxorange = '#F46036'\n",
    "# rustyredy = '#C2320A'\n",
    "\n",
    "# diff_gradient = [uberplant, eggplants, periptant, blueplant, kindablue, fullwhite, sadorange, tangerine, prforange, xxxorange, rustyredy]\n",
    "# selfish_gradient = [uberplant, eggplants, periptant, blueplant, kindablue, fullwhite]\n",
    "\n",
    "purpl = '#241e4e'\n",
    "blue1 = '#587792'\n",
    "blue2 = '#8db1ab'\n",
    "green = '#C7E576'\n",
    "yell1 = '#f2ed6f'\n",
    "yell2 = '#f4e04d'\n",
    "yell3 = '#F0D719'\n",
    "orang = '#ff4000'\n",
    "\n",
    "diff_gradient = [purpl,blue1, blue2, green, fullwhite, yell1, yell2, yell3, orang]\n",
    "selfish_gradient =[purpl, fullwhite]\n",
    "\n",
    "diff_cmap = LinearSegmentedColormap.from_list('interaction', diff_gradient)\n",
    "diff_cmap.set_bad(orang)\n",
    "diff_cmap.set_under(purpl)\n",
    "\n",
    "\n",
    "selfish_cmap = LinearSegmentedColormap.from_list('interaction', selfish_gradient)\n",
    "selfish_cmap.set_bad(orang)\n",
    "selfish_cmap.set_under(purpl)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3353d154a5e04200aa0d91b7b1e67382",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(VBox(children=(HBox(children=(Dropdown(index=2, options=('2L', '2R', '3L', '3R', '4', 'X', 'Y')…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n"
     ]
    }
   ],
   "source": [
    "# Browser for given ranges:\n",
    "directory = '/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation'\n",
    "comparison = 'JW18-DOX-1_JW18-wMel-1' \n",
    "resolution = '1kb'\n",
    "TEST_RANGE = '3L:2980000-3970000'\n",
    "# TEST_RANGE = '3R:31491078-31492678' \n",
    "JW18_DOX  = '#BAD86E'\n",
    "JW18_wMel = purpl\n",
    "#contact map 1\n",
    "file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "#Contact map2\n",
    "file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "#genes with open regions that are enriched in file1\n",
    "gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#genes with open regions that are enriched in file1\n",
    "gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#genes with open regions that are enriched in file1\n",
    "gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "#mustache identified de loops1\n",
    "loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "#mustache identified de loops2\n",
    "loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "\n",
    "###Tracks\n",
    "#RNA-coverage \n",
    "rnaseq_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18DOX221117.avg.coverage.bw'\n",
    "rnaseq_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117.avg.coverage.bw'\n",
    "#Micro-C_coverage \n",
    "microc_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-DOX-1.coverage.bw'\n",
    "microc_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel-1.coverage.bw'\n",
    "#Mappability \n",
    "mappability =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/dmel_6_genmap_e0_k150_final.bw'\n",
    "\n",
    "c1,c2=pairwise[comparison]\n",
    "\n",
    "\n",
    "with MinValue(-9), MaxValue(-1):\n",
    "    cool1 = Cool(file1)\n",
    "    cool2 = Cool(file2)\n",
    "\n",
    "frame = ChromName()+ XAxis() + \\\n",
    "    Selfish(cool2, cool1, norm=\"log\", cmap=selfish_cmap) +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "    MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "    HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=diff_cmap) + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "    MinValue(-6) + MaxValue(6)  + \\\n",
    "    TADCoverage(loops1, border_only=False, alpha=0.45, color=JW18_DOX,border_color=JW18_DOX) +\\\n",
    "    TADCoverage(loops2, border_only=False, alpha=0.25, color=JW18_wMel,border_color=JW18_wMel) +\\\n",
    "    GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "    GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "    BigWig(rnaseq_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-DOX-1\")+\\\n",
    "    BigWig(rnaseq_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-wMel-1\")+\\\n",
    "    BigWig(microc_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-DOX-1\")+\\\n",
    "    BigWig(microc_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-wMel-1\")+\\\n",
    "    BigWig(mappability) + Color('grey') + MinValue(0) + MaxValue(1)+ Title(\"Mappability\")+TrackHeight(1) +\\\n",
    "    XAxis()\n",
    "frame *= Feature(depth_ratio=0.35)\n",
    "frame.plot(TEST_RANGE)\n",
    "\n",
    "bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "bsr.goto(TEST_RANGE)\n",
    "bsr.show()\n",
    "bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "08a3f9bcca7d4eb5bbcbb571a78b313b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "VBox(children=(VBox(children=(HBox(children=(Dropdown(index=2, options=('2L', '2R', '3L', '3R', '4', 'X', 'Y')…"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop1.bed.bgz.tbi\n",
      "[W::hts_idx_load3] The index file is older than the data file: /scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/1kb/p_value_0.0001/bed/plot_bed/JW18-DOX-1_JW18-wMel-1.diffloop2.bed.bgz.tbi\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "File saved to: /scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/JW18-DOX-1_JW18-wMel-1_3L:2980000-3970000.svg\n"
     ]
    }
   ],
   "source": [
    "# Browser for given ranges:\n",
    "directory = '/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation'\n",
    "comparison = 'JW18-DOX-1_JW18-wMel-1' \n",
    "resolution = '1kb'\n",
    "TEST_RANGE = '3L:2980000-3970000'\n",
    "# TEST_RANGE = '3R:31491078-31492678' \n",
    "JW18_DOX  = '#BAD86E'\n",
    "JW18_wMel = purpl\n",
    "#contact map 1\n",
    "file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "#Contact map2\n",
    "file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "#genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#all genes\n",
    "gtf_all='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-r6.46.sorted.uniq.gtf'\n",
    "# # #genes with open regions that are enriched in file1\n",
    "# gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "# #genes with open regions that are enriched in file1\n",
    "# gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "#Genes enriched in both \n",
    "gtf3=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop.symbol.gtf'\n",
    "\n",
    "\n",
    "#mustache identified de loops1\n",
    "loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "#mustache identified de loops2\n",
    "loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "# print(f'cat {gtf1} {gtf2} |uniq > {gtf3}')\n",
    "\n",
    "###Tracks\n",
    "#RNA-coverage \n",
    "rnaseq_cov =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117_vs_DOX.diff.coverage.bw'\n",
    "# rnaseq_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18DOX221117.avg.coverage.bw'\n",
    "# rnaseq_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117.avg.coverage.bw'\n",
    "#Micro-C_coverage \n",
    "microc_cov =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel_vs_DOX.diff.coverage.bw'\n",
    "# microc_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-DOX-1.coverage.bw'\n",
    "# microc_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel-1.coverage.bw'\n",
    "#Mappability \n",
    "mappability =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/dmel_6_genmap_e0_k150_final.bw'\n",
    "\n",
    "c1,c2=pairwise[comparison]\n",
    "\n",
    "with MinValue(-9), MaxValue(-1):\n",
    "    cool1 = Cool(file1)\n",
    "    cool2 = Cool(file2)\n",
    "\n",
    "frame = ChromName()+ XAxis() + \\\n",
    "    Selfish(cool2, cool1, norm=\"log\", cmap=selfish_cmap) +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "    MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "    HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=diff_cmap) + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "    MinValue(-6) + MaxValue(6)  + \\\n",
    "    TADCoverage(loops1, border_only=False, alpha=0.45, color=JW18_DOX,border_color=JW18_DOX) +\\\n",
    "    TADCoverage(loops2, border_only=False, alpha=0.25, color=JW18_wMel,border_color=JW18_wMel) +\\\n",
    "    GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ TrackHeight(3)+\\\n",
    "    GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ TrackHeight(3)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(rnaseq_cov) + Color(JW18_wMel) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\nJW18-wMel enrichment\\nover JW18-DOX\")+TrackHeight(1)+\\\n",
    "    BigWig(rnaseq_cov) + Color(JW18_DOX) + MinValue(-500) + MaxValue(0)+ TrackHeight(1)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(microc_cov) + Color(JW18_wMel) + MinValue(0) + MaxValue(1000)+ Title(\"Micro-C coverage\\nJW18-wMel enrichment\\nover JW18-DOX\")+TrackHeight(1)+\\\n",
    "    BigWig(microc_cov) + Color(JW18_DOX) + MinValue(-1000) + MaxValue(0)+ TrackHeight(1)+\\\n",
    "    Spacer(height=.5) +\\\n",
    "    BigWig(mappability) + MinValue(0) + MaxValue(1)+ Title(\"Mappability\")+TrackHeight(1) +\\\n",
    "    XAxis()\n",
    "frame *= Feature(depth_ratio=0.35)\n",
    "frame.plot(TEST_RANGE)\n",
    "\n",
    "bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "bsr.goto(TEST_RANGE)\n",
    "bsr.show()\n",
    "bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.svg')\n",
    "print(f\"File saved to: {directory}/{comparison}_{TEST_RANGE}.svg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# # TEST_RANGE = '3L:2980000-3970000'\n",
    "\n",
    "# def plot_test_range(TEST_RANGE,folder):\n",
    "#     # Browser for given ranges:\n",
    "#     directory = f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/comparisons/plots/{folder}'\n",
    "#     comparison = 'JW18-DOX-1_JW18-wMel-1' \n",
    "#     resolution = '1kb'\n",
    "\n",
    "#     JW18_DOX  = '#BAD86E'\n",
    "#     JW18_wMel = purpl\n",
    "#     #contact map 1\n",
    "#     file1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][0]}.matrix_1kb.mcool'\n",
    "#     #Contact map2\n",
    "#     file2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/{pairwise[comparison][1]}.matrix_1kb.mcool'\n",
    "#     #all genes\n",
    "#     gtf_all='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-r6.46.sorted.gtf'\n",
    "#     #genes with open regions that are enriched in file1\n",
    "#     gtf1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop1.symbol.gtf'\n",
    "#     #genes with open regions that are enriched in file1\n",
    "#     gtf2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/gtf/symbol/{comparison}.diffloop2.symbol.gtf'\n",
    "#     #mustache identified de loops1\n",
    "#     loops1=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed'\n",
    "#     #mustache identified de loops2\n",
    "#     loops2=f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed'\n",
    "\n",
    "\n",
    "#     ###Tracks\n",
    "#     #RNA-coverage \n",
    "#     rnaseq_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18DOX221117.avg.coverage.bw'\n",
    "#     rnaseq_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/RNA-seq_coverage/JW18wMel221117.avg.coverage.bw'\n",
    "#     #Micro-C_coverage \n",
    "#     microc_cov_1 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-DOX-1.coverage.bw'\n",
    "#     microc_cov_2 =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/Micro-C_coverage/JW18-wMel-1.coverage.bw'\n",
    "#     #Mappability \n",
    "#     mappability =f'/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/dmel_6_genmap_e0_k150_final.bw'\n",
    "\n",
    "#     c1,c2=pairwise[comparison]\n",
    "\n",
    "#     with MinValue(-9), MaxValue(-1):\n",
    "#         cool1 = Cool(file1)\n",
    "#         cool2 = Cool(file2)\n",
    "\n",
    "#     frame = ChromName()+ XAxis() + \\\n",
    "#         Selfish(cool2, cool1, norm=\"log\", cmap=selfish_cmap) +Title(\"FDR Corrected\\np-value\")+ \\\n",
    "#         MinValue(0.000000001) + MaxValue(0.1)  + \\\n",
    "#         HiCDiff(cool2, cool1, normalize='log', diff_method='diff', cmap=diff_cmap) + Title(\"Differential\\nChromatin\\nInteractions\") + \\\n",
    "#         MinValue(-6) + MaxValue(6)  + \\\n",
    "#         TADCoverage(loops1, border_only=False, alpha=0.45, color=JW18_DOX,border_color=JW18_DOX) +\\\n",
    "#         TADCoverage(loops2, border_only=False, alpha=0.25, color=JW18_wMel,border_color=JW18_wMel) +\\\n",
    "#         GTF(gtf_all)+Title(f\"DmeL GTF\\n\")+ \\\n",
    "#         GTF(gtf1)+Title(f\"Genetic elements\\nenriched in\\n{c1}\")+ \\\n",
    "#         GTF(gtf2)+Title(f\"Genetic elements\\nenriched in\\n{c2}\")+ \\\n",
    "#         BigWig(rnaseq_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-DOX-1\")+\\\n",
    "#         BigWig(rnaseq_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(500)+ Title(\"RNA-seq coverage\\n JW18-wMel-1\")+\\\n",
    "#         BigWig(microc_cov_1) + Color(JW18_DOX) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-DOX-1\")+\\\n",
    "#         BigWig(microc_cov_2) + Color(JW18_wMel) + MinValue(0) + MaxValue(5000)+ Title(\"Micro-C coverage\\n JW18-wMel-1\")+\\\n",
    "#         BigWig(mappability) + Color('grey') + MinValue(0) + MaxValue(1)+ Title(\"Mappability\")+TrackHeight(1) +\\\n",
    "#         XAxis()\n",
    "#     frame *= Feature(depth_ratio=0.35)\n",
    "#     frame.plot(TEST_RANGE)\n",
    "\n",
    "#     bsr = Browser(frame, reference_genome='/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/reference/dmel-all-chromosome_only-r6.46.genome')\n",
    "#     bsr.goto(TEST_RANGE)\n",
    "#     # bsr.show()\n",
    "#     bsr.save(f'{directory}/{comparison}_{TEST_RANGE}.pdf')\n",
    "\n",
    "# import glob\n",
    "# import os\n",
    "\n",
    "# folder = '/scratch1/jodie/wolbachia/Wolbachia_induced_differentiation/comparisons'\n",
    "# file_paths = glob.glob(f\"{folder}/*.txt\")\n",
    "\n",
    "# # Extract filenames without path and extension\n",
    "# file_names = [os.path.splitext(os.path.basename(path))[0] for path in file_paths]\n",
    "\n",
    "# # Create folders based on these names\n",
    "# for name in file_names:\n",
    "#     # Define the directory path where you want to create the new folder\n",
    "#     new_dir_path = os.path.join(folder,'plots', name)\n",
    "#    # Create the directory (if it doesn't already exist)\n",
    "#     os.makedirs(new_dir_path, exist_ok=True)\n",
    "\n",
    "#     comparison = pd.read_csv(f'{folder}/{name}.txt', sep='\\t')\n",
    "#     for location in comparison['location']:\n",
    "#         # print(new_dir_path)\n",
    "#         plot_test_range(location,name)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# resolution='10kb'\n",
    "\n",
    "# for comparison in pairwise.keys():\n",
    "#     directory=f\"/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/plots/{resolution}/{comparison}\"\n",
    "#     if not os.path.exists(directory):\n",
    "#         # If the directory does not exist, create it\n",
    "#         os.makedirs(directory)\n",
    "#     ranges1=pd.read_csv(f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed', sep='\\t', header=None)\n",
    "#     ranges2=pd.read_csv(f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed', sep='\\t', header=None)\n",
    "#     ranges=pd.concat([ranges1,ranges2],axis=0)\n",
    "#     ranges.drop_duplicates()\n",
    "\n",
    "#     for index, row in ranges.iterrows():\n",
    "#     # Print the first, second, and third elements of each row\n",
    "#         TEST_RANGE=f\"{row[0]}:{floor(row[1]/1.1)}-{floor(row[2]*1.1)}\"\n",
    "        \n",
    "#         plot(TEST_RANGE,pairwise,comparison,resolution,directory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# resolution='1kb'\n",
    "\n",
    "# for comparison in pairwise.keys():\n",
    "#     directory=f\"/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/plots/{resolution}/{comparison}\"\n",
    "#     if not os.path.exists(directory):\n",
    "#         # If the directory does not exist, create it\n",
    "#         os.makedirs(directory)\n",
    "#     ranges1=pd.read_csv(f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop1.bed', sep='\\t', header=None)\n",
    "#     ranges2=pd.read_csv(f'/scratch1/jodie/wolbachia/Micro-C/09Nov2023_Micro_C/map/mustache_v1/{resolution}/p_value_0.0001/bed/plot_bed/{comparison}.diffloop2.bed', sep='\\t', header=None)\n",
    "#     ranges=pd.concat([ranges1,ranges2],axis=0)\n",
    "#     ranges.drop_duplicates()\n",
    "\n",
    "#     for index, row in ranges.iterrows():\n",
    "#     # Print the first, second, and third elements of each row\n",
    "#         TEST_RANGE=f\"{row[0]}:{floor(row[1]/1.1)}-{floor(row[2]*1.1)}\"\n",
    "        \n",
    "#         plot(TEST_RANGE,pairwise,comparison,resolution,directory)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
